src/Microsoft.ML.AutoML.Interactive/AutoMLKernelExtension.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Interactive;
+using Microsoft.DotNet.Interactive.Formatting;
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using XPlot.Plotly;
+
+namespace Microsoft.ML.AutoML.Interactive
+{
+    public class AutoMLKernelExtension : IKernelExtension
+    {
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async Task OnLoadAsync(Kernel kernel)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            Formatter.Register<NotebookMonitor>((monitor, writer) =>
+            {
+
+                var chart = Chart.Plot(
+                            new Scatter()
+                            {
+                                x = monitor.CompletedTrials.Select(x => x.TrialSettings.TrialId),
+                                y = monitor.CompletedTrials.Select(x => x.Metric),
+                                mode = "markers",
+                            }
+                        );
+
+                var layout = new Layout.Layout() { title = $"Plot metrics over trials." };
+                chart.WithLayout(layout);
+                chart.Width = 500;
+                chart.Height = 500;
+                chart.WithXTitle("Trial");
+                chart.WithYTitle("Metric");
+                chart.WithLegend(false);
+
+
+                var scriptJs = chart.GetInlineJS().Replace("<script>", String.Empty).Replace("</script>", String.Empty);
+
+
+
+                var trialHtml = String.Join(" ", monitor.CompletedTrials.Select(x => $@"
+		<tr>
+            <td>{x.TrialSettings.TrialId}</td>
+            <td>{x.Metric}</td>
+            <td>{x.TrialSettings.Pipeline}</td>
+        </tr>
+	"));
+
+
+
+                var bestRun = monitor.BestTrial == null ? "" :
+                $@"
+	<h3>Best Run</h3>
+	<p>
+		Trial: {monitor.BestTrial.TrialSettings.TrialId} <br>
+	</p>
+	";
+
+                var activeRunParam = monitor.ActiveTrial == null ? "" : JsonSerializer.Serialize(monitor.ActiveTrial.Parameter, new JsonSerializerOptions() { WriteIndented = true, });
```

Comment:
Careful with json serialization here,  we had to set the encoder in few places like suggested [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-character-encoding#serialize-all-characters)